File: DSA Implementations/Python/sorting_matrix.py
def sort_matrix(row, col, arr):
    for i in range(row):
        for j in range(col):
            for k in range(row):
                for l in range(col):
                    if arr[k][l] > arr[i][j]:
                        arr[i][j], arr[k][l] = arr[k][l], arr[i][j]

row = int(input("Enter the number of rows in the array : "))
col = int(input("Enter the number of columns in the array : "))
arr = []

print("Enter the elements of the array : ")
for _ in range(row):
    val = []
    for _ in range(col):
        val.append(int(input())) 
    arr.append(val)

sort_matrix(row, col, arr)

print("Sorted array : ")
print(*arr)

File: DSA Implementations/C++/sorting_matrix.cpp
#include <iostream>
#include <vector>
#include <algorithm>

void sort_matrix(std::vector<std::vector<int>> &arr)
{
    if (arr.empty() || arr[0].empty())
    {
        return;
    }

    int rows = arr.size();
    int cols = arr[0].size();

    std::vector<int> temp_vec;
    for (int i = 0; i < rows; ++i)
    {
        for (int j = 0; j < cols; ++j)
        {
            temp_vec.push_back(arr[i][j]);
        }
    }

    std::sort(temp_vec.begin(), temp_vec.end());

    int k = 0;
    for (int i = 0; i < rows; ++i)
    {
        for (int j = 0; j < cols; ++j)
        {
            arr[i][j] = temp_vec[k++];
        }
    }
}

int main()
{
    int rows, cols;

    std::cout << "Enter the number of rows in the matrix: ";
    std::cin >> rows;

    std::cout << "Enter the number of columns in the matrix: ";
    std::cin >> cols;

    std::vector<std::vector<int>> arr(rows, std::vector<int>(cols));

    std::cout << "Enter the elements of the matrix: " << std::endl;
    for (int i = 0; i < rows; ++i)
    {
        for (int j = 0; j < cols; ++j)
        {
            std::cin >> arr[i][j];
        }
    }

    sort_matrix(arr);

    std::cout << "Sorted matrix: " << std::endl;
    for (int i = 0; i < rows; ++i)
    {
        for (int j = 0; j < cols; ++j)
        {
            std::cout << arr[i][j] << " ";
        }
        std::cout << std::endl;
    }

    return 0;
}